drivers/sensor/iis3dhhc/iis3dhhc.c

```diff
@@ -207,34 +207,38 @@ static int iis3dhhc_init(const struct device *dev)
 	}
 
 #ifdef CONFIG_IIS3DHHC_TRIGGER
-	if (iis3dhhc_init_interrupt(dev) < 0) {
-		LOG_ERR("Failed to initialize interrupt.");
-		return -EIO;
+	if (config->int_gpio.port) {
+		if (iis3dhhc_init_interrupt(dev) < 0) {
+			LOG_ERR("Failed to initialize interrupt.");
+			return -EIO;
+		}
 	}
 #endif
 
 	return 0;
 }
 
-static struct iis3dhhc_data iis3dhhc_data;
-
-static const struct iis3dhhc_config iis3dhhc_config = {
-#ifdef CONFIG_IIS3DHHC_TRIGGER
-#ifdef CONFIG_IIS3DHHC_DRDY_INT1
```

Comment:
are these Kconfig options needed anymore?

Reply:
I guess the logic of this driver can be improved later. Now you need 2 `irq-gpios` in DT, even though only 1 is used by the driver, depending on a Kconfig option that applies to all instances. If I remember correctly, there was another driver that was using `int-pin` to specify wether to use `INT1` or `INT2`. Do you prefer to fix this problem here or follow-up PR?

Reply:
Agreed, that's not very convenient. Will try to fix this in a follow-up PR.